src/env_universal_common.rs

```diff
@@ -750,23 +746,19 @@ impl EnvUniversal {
     fn save(&mut self, directory: &wstr) -> bool {
         assert!(self.ok_to_save, "It's not OK to save");
 
-        let mut private_file_path = WString::new();
-
         // Open adjacent temporary file.
-        let private_fd = self.open_temporary_file(directory, &mut private_file_path);
-        let mut success = private_fd.is_valid();
+        let private_fd = self.open_temporary_file(directory);
 
-        if !success {
+        let Ok((private_fd, private_file_path)) = private_fd else {
             FLOG!(uvar_file, "universal log open_temporary_file() failed");
-        }
+            // We can early return because file creation failed, so we don't have to clean it up
+            return false;
```

Comment:
This is also a change in behavior, previously if `mkstemp` /`mkostemp` failed we called `unlink` on whatever was left in the template buffer, but as mentioned in the comment above, it seems that contents of that buffer after error occurs are not defined.

So here I'm assuming that mkstemp failure does not require cleanup.